diff -ruN a/plugin/__init__.py b/plugin/__init__.py
--- a/plugin/__init__.py	2022-07-20 21:10:51.000000000 +0300
+++ b/plugin/__init__.py	2023-09-23 15:41:54.388260930 +0300
@@ -0,0 +1,16 @@
+from Components.Language import language
+from Tools.Directories import resolveFilename, SCOPE_PLUGINS, SCOPE_LANGUAGE
+from os import environ as os_environ
+import gettext
+
+def localeInit():
+	gettext.bindtextdomain("AddStreamUrl", resolveFilename(SCOPE_PLUGINS, "Extensions/AddStreamUrl/locale"))
+
+def _(txt):
+	t = gettext.dgettext("AddStreamUrl", txt)
+	if t == txt:
+		t = gettext.gettext(txt)
+	return t
+
+localeInit()
+language.addCallback(localeInit)
diff -ruN a/plugin/plugin.py b/plugin/plugin.py
--- a/plugin/plugin.py	2023-09-23 14:53:28.007605073 +0300
+++ b/plugin/plugin.py	2023-09-23 14:34:54.847367212 +0300
@@ -1,3 +1,4 @@
+from . import _
 from Plugins.Plugin import PluginDescriptor
 from Screens.Screen import Screen
 from Screens.MessageBox import MessageBox
@@ -8,6 +9,7 @@
 from Screens.VirtualKeyBoard import VirtualKeyBoard
 from urllib.parse import quote
 from enigma import eDVBDB
+import os
 
 class LiveStreamingLinksHeader(Screen):
 	skin = """
diff -ruN a/po/AddStreamUrl.pot b/po/AddStreamUrl.pot
--- a/po/AddStreamUrl.pot	1970-01-01 03:00:00.000000000 +0300
+++ b/po/AddStreamUrl.pot	2023-09-23 14:34:54.847367212 +0300
@@ -0,0 +1,42 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+#, fuzzy
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2022-10-15 12:34+0300\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
+"Language-Team: LANGUAGE <LL@li.org>\n"
+"Language: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=CHARSET\n"
+"Content-Transfer-Encoding: 8bit\n"
+
+msgid "Add a streaming url to your channellist"
+msgstr ""
+
+msgid "Add stream URL"
+msgstr ""
+
+msgid "Cancel"
+msgstr ""
+
+msgid "Enter URL"
+msgstr ""
+
+msgid "Enter name"
+msgstr ""
+
+msgid "Ok"
+msgstr ""
+
+msgid "Select URL type"
+msgstr ""
+
+msgid "Select a bouquet to add a channel to"
+msgstr ""
diff -ruN a/po/ru.po b/po/ru.po
--- a/po/ru.po	1970-01-01 03:00:00.000000000 +0300
+++ b/po/ru.po	2023-09-23 14:34:54.847367212 +0300
@@ -0,0 +1,43 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: AddStreamUrl\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2022-10-15 12:34+0300\n"
+"PO-Revision-Date: 2022-10-15 12:43+0300\n"
+"Last-Translator: 1455\n"
+"Language-Team: \n"
+"Language: ru\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=3; plural=(n%10==1 && n%100!=11 ? 0 : n%10>=2 && n%10<=4 && (n%100<12 || n%100>14) ? 1 : 2);\n"
+"X-Generator: Poedit 3.0.1\n"
+
+msgid "Add a streaming url to your channellist"
+msgstr "Добавка ссылки на трансляцию в свой список каналов"
+
+msgid "Add stream URL"
+msgstr "Добавить URL стрима"
+
+msgid "Cancel"
+msgstr "Отмена"
+
+msgid "Enter URL"
+msgstr "Ввод URL"
+
+msgid "Enter name"
+msgstr "Введите имя"
+
+msgid "Ok"
+msgstr "ОК"
+
+msgid "Select URL type"
+msgstr "Выбор URL-типа"
+
+msgid "Select a bouquet to add a channel to"
+msgstr "Выбор и добавление буклета"
diff -ruN a/setup.py b/setup.py
--- a/setup.py	2023-09-23 14:53:32.283677983 +0300
+++ b/setup.py	2023-09-23 15:39:53.021732296 +0300
@@ -1,10 +1,13 @@
 from distutils.core import setup
+import setup_translate
 
 pkg = 'Extensions.AddStreamUrl'
 setup (name = 'enigma2-plugin-extensions-addstreamurl',
 	version = '1.0',
 	description = 'Add a stream url to channel list',
 	packages = [pkg],
-	package_dir = {pkg: 'plugin'}
-)
+	package_dir = {pkg: 'plugin'},
+	package_data = {pkg: ['*.*']},
+	cmdclass = setup_translate.cmdclass, # for translation
+	)
 
diff -ruN a/setup_translate.py b/setup_translate.py
--- a/setup_translate.py	1970-01-01 03:00:00.000000000 +0300
+++ b/setup_translate.py	2023-09-23 14:34:54.847367212 +0300
@@ -0,0 +1,51 @@
+
+# Language extension for distutils Python scripts. Based on this concept:
+# http://wiki.maemo.org/Internationalize_a_Python_application
+from distutils import cmd
+from distutils.command.build import build as _build
+import glob
+import os
+
+
+class build_trans(cmd.Command):
+	description = 'Compile .po files into .mo files'
+
+	def initialize_options(self):
+		pass
+
+	def finalize_options(self):
+		pass
+
+	def run(self):
+		s = os.path.join('po')
+		lang_domains = glob.glob(os.path.join(s, '*.pot'))
+		if len(lang_domains):
+			for lang in os.listdir(s):
+				if lang.endswith('.po'):
+					plugin = os.path.join(s, lang)
+					lang = lang[:-3]
+					destdir = os.path.join('build', 'lib', 'Extensions',
+						'AddStreamUrl', 'locale', lang, 'LC_MESSAGES')
+					if not os.path.exists(destdir):
+						os.makedirs(destdir)
+					for lang_domain in lang_domains:
+						lang_domain = lang_domain.rsplit('/', 1)[1]
+						dest = os.path.join(destdir, lang_domain[:-3] + 'mo')
+						print("Language compile %s -> %s" % (plugin, dest))
+						if os.system("msgfmt '%s' -o '%s'" % (plugin, dest)) != 0:
+							raise Exception("Failed to compile: " + plugin)
+		else:
+			print("we got no domain -> no translation was compiled")
+
+
+class build(_build):
+	sub_commands = _build.sub_commands + [('build_trans', None)]
+
+	def run(self):
+		_build.run(self)
+
+
+cmdclass = {
+	'build': build,
+	'build_trans': build_trans,
+}
