diff -ruN a/plugin/__init__.py b/plugin/__init__.py
--- a/plugin/__init__.py	2023-07-20 11:20:29.000000000 +0300
+++ b/plugin/__init__.py	2023-07-20 11:31:50.171746515 +0300
@@ -1,4 +1,3 @@
-# -*- coding: utf-8 -*-
 
 from Components.Language import language
 from Tools.Directories import resolveFilename, SCOPE_PLUGINS
diff -ruN a/plugin/keymap.xml b/plugin/keymap.xml
--- a/plugin/keymap.xml	2023-07-20 11:20:29.000000000 +0300
+++ b/plugin/keymap.xml	2023-07-20 11:25:59.110153586 +0300
@@ -26,7 +26,7 @@
 		<key id="KEY_8" mapto="8" flags="m" />
 		<key id="KEY_9" mapto="9" flags="m" />
 		<key id="KEY_0" mapto="0" flags="m" />
-		                                       
+
 		<key id="KEY_NEXT" mapto="next" flags="m" />
 		<key id="KEY_PREVIOUS" mapto="previous" flags="m" />
 
@@ -42,6 +42,6 @@
 		<key id="KEY_HOME" mapto="cancel" flags="m" />
 		<key id="KEY_INFO" mapto="showEventInfo" flags="m" />
 		<key id="KEY_HELP" mapto="contextMenu" flags="m" />
-		<key id="\x1b" mapto="cancel" flags="m" />
+		<key id="KEY_ESC" mapto="cancel" flags="m" />
 	</map>
 </keymap>
diff -ruN a/plugin/locale/ru/LC_MESSAGES/Foreca.po b/plugin/locale/ru/LC_MESSAGES/Foreca.po
--- a/plugin/locale/ru/LC_MESSAGES/Foreca.po	2023-07-20 11:20:29.000000000 +0300
+++ b/plugin/locale/ru/LC_MESSAGES/Foreca.po	2023-07-20 11:25:59.110153586 +0300
@@ -234,7 +234,7 @@
 
 #: ../plugin.py:21 ../plugin.py:27 ../ui.py:635 ../ui.py:654
 msgid "Foreca Weather Forecast"
-msgstr "Foreca - прогноз погоды"
+msgstr "'Foreca' - Прогноз погоды"
 
 #: ../ui.py:1124
 msgid "Forecast"
diff -ruN a/plugin/ui.py b/plugin/ui.py
--- a/plugin/ui.py	2023-07-20 11:20:29.000000000 +0300
+++ b/plugin/ui.py	2023-07-20 11:37:59.997728523 +0300
@@ -1,6 +1,3 @@
-# -*- coding: UTF-8 -*-
-from __future__ import absolute_import
-from __future__ import print_function
 VERSION = "3.2.8"
 #
 #  $Id$
@@ -117,7 +114,7 @@
 
 # for localized messages
 from . import _
-from locale import setlocale, LC_COLLATE, strxfrm
+from .locale import setlocale, LC_COLLATE, strxfrm
 
 # PYTHON IMPORTS
 from os import makedirs, unlink
@@ -153,8 +150,8 @@
 if PY3:
 	from urllib.request import urlopen, Request, pathname2url
 else:
-	from urllib import pathname2url
-	from urllib2 import urlopen, Request
+	from urllib.request import pathname2url
+	from urllib.request import urlopen, Request
 
 pluginPrintname = "[Foreca Ver. %s]" % VERSION
 ###############################################################################
diff -ruN a/setup_translate.py b/setup_translate.py
--- a/setup_translate.py	2023-07-20 11:20:29.000000000 +0300
+++ b/setup_translate.py	2023-07-20 11:33:59.765899348 +0300
@@ -1,4 +1,3 @@
-from __future__ import print_function
 # Language extension for distutils Python scripts. Based on this concept:
 # http://wiki.maemo.org/Internationalize_a_Python_application
 from distutils import cmd
@@ -27,7 +26,7 @@
 						dest = join(lc, f[:-2] + 'mo')
 						print("Language compile %s -> %s" % (src, dest))
 						if system("msgfmt '%s' -o '%s'" % (src, dest)) != 0:
-							raise (Exception, "Failed to compile: " + src)
+							raise Exception
 
 
 class build(_build):
