diff -ruN a/src/keymap.xml b/src/keymap.xml
--- a/src/keymap.xml	2022-10-10 14:47:37.636418148 +0300
+++ b/src/keymap.xml	2022-10-11 12:25:41.296253635 +0300
@@ -1,19 +1,19 @@
 <keymap>
 	<map context="SatellitesEditorActions">
 		<key id="KEY_OK" mapto="select" flags="m" />
-		<key id="KEY_EXIT" mapto="exit" flags="m" />
-		
+		<key id="KEY_ESC" mapto="exit" flags="m" />
+
 		<key id="KEY_LEFT" mapto="left" flags="m" />
 		<key id="KEY_RIGHT" mapto="right" flags="m" />
-		
+
 		<key id="KEY_UP" mapto="up" flags="m" />
 		<key id="KEY_UP" mapto="upUp" flags="b" />
 		<key id="KEY_UP" mapto="upRepeated" flags="r" />
-		
+
 		<key id="KEY_DOWN" mapto="down" flags="m" />
 		<key id="KEY_DOWN" mapto="downUp" flags="b" />
 		<key id="KEY_DOWN" mapto="downRepeated" flags="r" />
-		
+
 		<key id="KEY_RED" mapto="red" flags="m" />
 		<key id="KEY_RED" mapto="redRepeated" flags="r" />
 		<key id="KEY_RED" mapto="redUp" flags="b" />
@@ -22,10 +22,10 @@
 		<key id="KEY_GREEN" mapto="greenUp" flags="b" />
 		<key id="KEY_YELLOW" mapto="yellow" flags="m" />
 		<key id="KEY_BLUE" mapto="blue" flags="m" />
-		
+
 		<key id="KEY_CHANNELUP" mapto="nextPage" flags="m" />
 		<key id="KEY_CHANNELDOWN" mapto="prevPage" flags="m" />
-		
+
 		<key id="KEY_HELP" mapto="displayHelp" flags="m" />
 		<key id="KEY_MENU" mapto="displayMenu" flags="m" />
 		<key id="KEY_INFO" mapto="displayInfo" flags="m" />
diff -ruN a/src/plugin.py b/src/plugin.py
--- a/src/plugin.py	2022-10-10 14:47:37.636418148 +0300
+++ b/src/plugin.py	2022-10-10 21:48:27.421459936 +0300
@@ -1,5 +1,4 @@
-#!/usr/bin/python
-# -*- coding: utf-8 -*-
+#!/usr/bin/python3
 
 from . import _
 from Components.NimManager import nimmanager
diff -ruN a/src/satedit.py b/src/satedit.py
--- a/src/satedit.py	2022-10-10 14:47:37.640418148 +0300
+++ b/src/satedit.py	2022-10-10 21:55:40.069461654 +0300
@@ -1,6 +1,3 @@
-from __future__ import print_function
-from __future__ import division
-
 from . import _
 from Components.ActionMap import ActionMap
 from Components.Button import Button
@@ -178,7 +175,7 @@
                         print('transponderDoctor: Transponderdaten muessen vom Type DICT sein')
                         print(transponder)
                         return
-                param = transponder.keys()
+                param = list(transponder.keys())
                 transParam = {}
                 for x in param:
                         transParam[x] = x.lower()
@@ -995,7 +992,7 @@
 
         def createConfig(self):
                 self.configSatelliteName = ConfigText(default=self.satelliteName, visible_width=50, fixed_size=False)
-                self.configSatelliteName.setUseableChars(u"0,?!&@=*'+\()$~% 1.:;/-_#abc2ABCdef3DEFghi4GHIjkl5JKLmno6MNOpqrs7PQRStuv8TUVwxyz9WXYZ")
+                self.configSatelliteName.setUseableChars("0,?!&@=*'+\()$~% 1.:;/-_#abc2ABCdef3DEFghi4GHIjkl5JKLmno6MNOpqrs7PQRStuv8TUVwxyz9WXYZ")
                 self.configSatellitePosition = ConfigFloat(default=self.satellitePosition, limits=[(0, 180), (0, 9)])
                 self.configSatelliteOrientation = ConfigSelection([('east', 'East'), ('west', 'West')], self.satelliteOrientation)
                 options = self.satelliteFlags in (1, 3, 5, 7) and True or False
